app/models/concerns/non_null_session_uuid.rb

```diff
@@ -1,11 +0,0 @@
-module NonNullSessionUuid
-  extend ActiveSupport::Concern
-
-  included do
-    before_create :generate_session_uuid
-  end
-
-  def generate_session_uuid
-    self.session_uuid = '_' + SecureRandom.uuid
-  end
-end
```

Comment:
This is all related to this change. If this were in a separate PR, it would feel like an afterthought because the "Why" for that PR would basically be "to clean up the previous PR". The removal of this code only makes sense in the context of this PR.

Another way to put it is that if we had these tests in place when we first updated the saml_idp gem, they would have failed, and we would have had to make this change. After adding the tests in this PR, in order to make them pass, I had to remove the underscore from the UUID.